README.md

```diff
@@ -15,6 +15,55 @@ Our documentation covers:
 * [Failures that Kani can spot](https://model-checking.github.io/kani/tutorial-kinds-of-failure.html)
 * [Kani's current limitations](https://model-checking.github.io/kani/limitations.html)
 
+## What guarantees does Kani provide?
+If Kani verifies your code under a proof harness, then, under the assumptions of the specification, you have assurance that all code reachable from the proof harness:
+* Does not have runtime panics 
+* All user-added assertions are correct.
+* There are no memory safety issues, including in unsafe code.
+
+Adding support for additional checks is prioritized based on customer need.
+[We would appreciate your feedback as to which are the highest priority areas for your projects](https://github.com/model-checking/kani/issues/new/choose).
+
+## What do I need to trust when running Kani?
+Kani analyzes the MIR representation of a Rust program.
+If the real-world behaviour of the program diverges from the MIR, then Kani's guarantees may not hold.
+Examples of how MIR can differ from real-world behaviour are discussed below. 
+
+Kani uses the program analysis tool [CBMC](https://github.com/diffblue/cbmc) as a back-end.
+If the interpretation of MIR using CBMC diverges from the real-world behaviour of the MIR, then Kani's guarantees may not hold.
+Examples of how Kani may fail to faithfully model MIR behaviour are also discussed below.
+
+In particular, you need to trust the following stages of the Kani verification pipeline:
+
+1. **You need to trust the correctness of your specifications.**
+Kani analyzes code against a user specification, typically given as a “proof harness”.
+The assurance given by Kani depends on the specification correctly describing the environment in which the code runs, and the expected behavior of the code.
+If the specification has bugs, then Kani may (accurately) declare that buggy code is OK, since it complies with the buggy specification.
+We have developed training materials and checklists to aid developers in writing sound specifications for code-level model-checking (available at
+https://github.com/awslabs/aws-templates-for-cbmc-proofs/tree/master/training-material).
+Currently, the examples focus on C verification; we are working on adding Rust examples.
+
+1. **You need to trust the rustc compiler to properly parse rust into MIR.**
+Kani analyzes code at the MIR level.
+It uses the standard rustc compiler to translate Rust into MIR.
+Correctness guarantees, therefore, apply only to the particular compiler, and set of compilation flags (e.g. optimizations, machine architecture, etc) used.
+A different compiler, or the same compiler with different flags, may produce semantically different MIR from the same underlying source code.
+This can occur even in cases where there are no bugs in the underlying Rustc compiler, as parts of the Rust language are either undefined in the standard, or called out as implementation specific (for example, object layouts).
```

Comment:
```suggestion
This can occur even in cases where there are no bugs in the underlying Rustc compiler, as parts of the Rust language are either undefined in the standard, or called out as implementation specific (for example, object layouts).
```
```suggestion
This can occur even in cases where there are no bugs in the underlying Rustc compiler, as parts of the Rust language are either undefined in the standard, or called out as implementation-specific (for example, object layouts).
```